jme3-core/src/main/java/com/jme3/light/LightList.java

```diff
@@ -230,37 +231,53 @@ public void sort(boolean transformChanged) {
      * @param parent the parent's world-space LightList
      */
     public void update(LightList local, LightList parent) {
+        update(local, parent, null);
+    }
+
+
+    /**
+     * Updates a "world-space" light list, using the spatial's local-space
+     * light list, its parent's world-space light list and an optional filter.
+     *
+     * @param local the local-space LightList (not null)
+     * @param parent the parent's world-space LightList
+     * @param filter an optional filter to apply to the lights
+     *               (null means no filtering)
+     */
+    public void update(LightList local, LightList parent, Predicate<Light> filter) {
```

Comment:
added a filter here so that we can exclude global lights from being added the children's internal world lights list.

That's because the rootNode propagates down all the global lights, so when a global light is owned by a children and it is propagated to the same children later, we would end up with the children having two references to the same light, one from its internal local light list and one from the rootNode global light list.
To fix that we would need to deduplicate the lights in the list, that is not great for performances, so instead i've decided to use this filter based approach